### PR TITLE
network: type annotations for PeerId, NodeId and TrackerId

### PR DESCRIPTION
**Pull request by teogeb**
Added type aliases for `PeerId`, `NodeId` and `TrackerId`. 

All those types are `strings`, but as they have aliases, it is now easier to understand some of the data structures. E.g.
```
private readonly disconnectionTimers: { [key: string]: NodeJS.Timeout }
private readonly trackerBook: { [key: string]: string } // address => id
private readonly consecutiveDeliveryFailures: { [key: string]: number } // id => counter
```
are now 
```
private readonly disconnectionTimers: Record<NodeId,NodeJS.Timeout>
private readonly trackerBook: { [key: string]: TrackerId } // address => id
private readonly consecutiveDeliveryFailures: Record<NodeId,number> // id => counter
```

Renamed also some variables e.g. in `InstructionRetryManager` to reflect the types more accurately.

### CODE REVIEW

File: packages/network/src/connection/WebRtcEndpoint.ts [hunk 1]
```diff
@@ -120,7 +120,7 @@ export class WebRtcEndpoint extends EventEmitter implements IWebRtcEndpoint {
     }
 
     private createConnection(
-        targetPeerId: string,
+        targetPeerId: PeerId,
         routerId: string,
```
Comment (juslesan) [inline packages/network/src/connection/WebRtcEndpoint.ts:124]: RouterId could be annotated as signalerId, but a I prefer the more generic string type for now

File: packages/network/src/identifiers.ts [hunk 2]
```diff
@@ -50,13 +51,11 @@ export interface Location {
...
-        inboundNodes: string[]
-        outboundNodes: string[]
-        counter: number
-    }
-}
+export type StatusStreams = Record<StreamKey, {
+    inboundNodes: NodeId[]
+    outboundNodes: NodeId[]
+    counter: number
+}>
```
Comment (juslesan) [inline packages/network/src/identifiers.ts:58]: 👍

File: packages/network/src/protocol/NodeToNode.ts [hunk 3]
```diff
@@ -44,26 +45,26 @@ export class NodeToNode extends EventEmitter {
     }
 
     connectToNode(
-        receiverNodeId: string,
+        receiverNodeId: NodeId,
         trackerAddress: string,
         trackerInstructed = true
-    ): Promise<string> {
+    ): Promise<PeerId> {
```
Comment (juslesan) [inline packages/network/src/protocol/NodeToNode.ts]: NodeId is returned here

File: packages/network/src/protocol/TrackerNode.ts [hunk 4]
```diff
@@ -134,11 +136,11 @@ export class TrackerNode extends EventEmitter {
         return requestId
     }
 
-    async send<T>(receiverNodeId: string, message: T & TrackerLayer.TrackerMessage): Promise<void> {
-        await this.endpoint.send(receiverNodeId, message.serialize())
+    async send<T>(receiverPeerId: PeerId, message: T & TrackerLayer.TrackerMessage): Promise<void> {
```
Comment (juslesan) [inline packages/network/src/protocol/TrackerNode.ts]: Just a thought, the receiver here should always be a tracker, see second parameter type. However, once we move towards a decentralized tracker algorithm the receiver would be a super node (or a node with a public ip)

Given that in most cases in this class the "PeerId" is "TrackerId" maybe it's the better option for now.

File: packages/network/src/protocol/TrackerNode.ts [hunk 5]
```diff
@@ -157,7 +159,7 @@ export class TrackerNode extends EventEmitter {
         }
     }
 
-    connectToTracker(trackerAddress: string, trackerPeerInfo: PeerInfo): Promise<string> {
+    connectToTracker(trackerAddress: string, trackerPeerInfo: PeerInfo): Promise<PeerId> {
```
Comment (juslesan) [inline packages/network/src/protocol/TrackerNode.ts]: Always returns a TrackerId

File: packages/network/src/protocol/NodeToNode.ts [hunk 6]
```diff
@@ -117,7 +118,7 @@ export class NodeToNode extends EventEmitter {
         return this.endpoint.getRtts()
     }
 
-    getNegotiatedProtocolVersionsOnNode(peerId: string): [number, number] {
+    getNegotiatedProtocolVersionsOnNode(peerId: PeerId): [number, number] {
```
Comment (juslesan) [inline packages/network/src/protocol/NodeToNode.ts]: Change peerId: PeerId to nodeId: NodeId

**Review by juslesan — CHANGES_REQUESTED** [review]
In general, I think that it's best to avoid using the "PeerId" type in the Logic + Protocol layers. Instead use "NodeId" and "TrackerId" as much as possible there. In the connection layer, always using "PeerId" is fine as it is agnostic to the higher layer functionality.

File: packages/network/src/protocol/NodeToNode.ts (hunk 3 above)
Comment (teogeb) [reply packages/network/src/protocol/NodeToNode.ts]: Fixed

File: packages/network/src/protocol/NodeToNode.ts (hunk 6 above)
Comment (teogeb) [reply packages/network/src/protocol/NodeToNode.ts]: Fixed

File: packages/network/src/protocol/TrackerNode.ts (hunk 5 above)
Comment (teogeb) [reply packages/network/src/protocol/TrackerNode.ts]: Fixed

File: packages/network/src/protocol/TrackerNode.ts (hunk 4 above)
Comment (teogeb) [reply packages/network/src/protocol/TrackerNode.ts]: Changed to `TrackerId`

**Review by juslesan — APPROVED** [review]

**Pull request merged**